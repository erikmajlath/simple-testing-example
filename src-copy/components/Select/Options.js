import React from 'react'

export default ({ options, value, onChange }) =>
  <ul>
    {options.map(item => {
      const selected = value.includes(item.value)
      return (
        <li key={item.value} onClick={() => onChange(item, selected)}>
          {item.label}
          <span>
            {selected && ' - SELECTED!'}
          </span>
        </li>
      )
    })}
  </ul>

// Constants
export const SF = { value: 'sf', label: 'San Francisco' }
export const LA = { value: 'la', label: 'Los Angeles' }
export const NY = { value: 'ny', label: 'New York City' }

export const OPTIONS = [SF, LA, NY]
