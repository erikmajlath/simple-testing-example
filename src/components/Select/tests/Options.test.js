import React from 'react'
import Options, { SF, NY } from '../Options'
import { shallow, mount } from 'enzyme'

describe('Options', () => {

  it('renders items from options', () => {
    const wrapper = shallow(<Options options={[SF, NY]} value={[]} />)

    expect(wrapper.find('li').length).toBe(2)
  })

  it('matches snapshot', () => {
    const wrapper = shallow(<Options options={[SF, NY]} value={[]} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot with selected value', () => {
    const wrapper = shallow(<Options options={[SF, NY]} value={['sf']} />)

    expect(wrapper).toMatchSnapshot()
  })
})

