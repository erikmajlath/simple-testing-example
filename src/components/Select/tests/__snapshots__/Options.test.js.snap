// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options matches snapshot 1`] = `
<ul>
  <li
    onClick={[Function]}
  >
    San Francisco
    <span />
  </li>
  <li
    onClick={[Function]}
  >
    New York City
    <span />
  </li>
</ul>
`;

exports[`Options matches snapshot with selected value 1`] = `
<ul>
  <li
    onClick={[Function]}
  >
    San Francisco
    <span>
       - SELECTED!
    </span>
  </li>
  <li
    onClick={[Function]}
  >
    New York City
    <span />
  </li>
</ul>
`;
